src/isar/services/utilities/scheduling_utilities.py

```diff
@@ -24,28 +30,192 @@ class SchedulingUtilities:
+            self.logger.error(e)
+            raise HTTPException(
+                status_code=HTTPStatus.INTERNAL_SERVER_ERROR,
+                detail="Could not plan mission",
+            )
+
+    def verify_robot_capable_of_mission(
+        self, mission: Mission, robot_capabilities: List[str]
+    ) -> bool:
+        """Verify that the robot is capable of performing this mission
+
+        Raises
+        ------
+        HTTPException 400 Bad request
+            If the robot is not capable of performing mission
+        """
+        is_capable: bool = True
+        missing_capabilities: Set[str] = set()
+        for task in mission.tasks:
+            for step in task.steps:
+                if not step.type in robot_capabilities:
+                    is_capable = False
+                    missing_capabilities.add(step.type)
+
+        if not is_capable:
+            error_message = (
+                f"Bad Request - Robot is not capable of performing mission."
+                f" Missing functionalities: {missing_capabilities}."
+            )
+            self.logger.error(error_message)
+            raise HTTPException(
+                status_code=HTTPStatus.BAD_REQUEST,
+                detail=error_message,
+            )
+
+        return is_capable
+
+    def verify_state_machine_ready_to_receive_mission(self, state: States) -> bool:
+        """Verify that the state machine is idle and ready to receive a mission
+
+        Raises
+        ------
+        HTTPException 409 Conflict
+            If state machine is not idle and therefore can not start a new mission
+        """
+        is_state_machine_ready_to_receive_mission = state == States.Idle
+        if not is_state_machine_ready_to_receive_mission:
+            error_message = f"Conflict - Mission already in progress - State: {state}"
+            self.logger.warning(error_message)
+            raise HTTPException(status_code=HTTPStatus.CONFLICT, detail=error_message)
+
+        return is_state_machine_ready_to_receive_mission
 
     def start_mission(self, mission: Mission, initial_pose: Optional[Pose]) -> None:
-        self._send_command(
-            StartMissionMessage(mission=deepcopy(mission), initial_pose=initial_pose),
-            self.queues.start_mission,
-        )
+        """Start mission
+
+        Raises
+        ------
+        HTTTPException 408 Request timeout
+            If there is a timeout while communicating with the state machine
+        """
+        try:
+            self._send_command(
+                StartMissionMessage(
+                    mission=deepcopy(mission), initial_pose=initial_pose
+                ),
+                self.queues.start_mission,
+            )
+        except QueueTimeoutError:
+            error_message = "Timeout - Failed to start mission in ISAR"
+            self.logger.error(error_message)
+            raise HTTPException(
+                status_code=HTTPStatus.REQUEST_TIMEOUT, detail=error_message
```

Comment:
408 Request timeout implies a user error (all 400 codes do), I believe this should be a 500 internal server error as it has nothing to do with the user input

Reply:
Good catch, changed to 500 Internal server error